### PR TITLE
Humble

### PR DESCRIPTION
**Pull request by PranavDhulipala**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | (add tickets here #1) |
| Primary OS tested on | (Ubuntu, MacOS, Windows) |
| Robotic platform tested on | (Steve's Robot, gazebo simulation of Tally, hardware turtlebot) |

---

## Description of documentation updates required from your changes

---

### CODE REVIEW

File: nav2_smoother/test/test_simple_smoother.cpp [hunk 1]
```diff
@@ -158,7 +158,7 @@ TEST(SmootherTest, test_simple_smoother)
   straight_regular_path.poses[10].pose.position.y = 1.0;
 
   EXPECT_TRUE(smoother->smooth(straight_regular_path, max_time));
-  for (uint i = 0; i != straight_regular_path.poses.size() - 1; i++) {
+  for (unsigned int i = 0; i != straight_regular_path.poses.size() - 1; i++) {
```
Comment (ooeygui) [inline nav2_smoother/test/test_simple_smoother.cpp]: uint -> size_t or uint32_t

File: nav2_constrained_smoother/test/test_smoother_cost_function.cpp [hunk 2]
```diff
@@ -78,7 +78,7 @@ TEST_F(Test, testingCurvatureResidual)
   EXPECT_EQ(fn.getCurvatureResidual(0.0, pt, pt_other, pt_other), 0.0);
 
   nav2_constrained_smoother::SmootherParams params_no_min_turning_radius;
-  params_no_min_turning_radius.max_curvature = 1.0f / 0.0;
+  params_no_min_turning_radius.max_curvature = 1.0;
```
Comment (ooeygui) [inline nav2_constrained_smoother/test/test_smoother_cost_function.cpp]: std::nan("")

File: nav2_dwb_controller/nav_2d_utils/test/path_ops_test.cpp [hunk 3]
```diff
@@ -63,7 +63,7 @@ TEST(path_ops_test, AdjustResolutionSimple)
 
   nav_2d_msgs::msg::Path2D out = adjustPlanResolution(in, RESOLUTION);
   float length = 100;
-  ulong number_of_points = ceil(length / (2 * RESOLUTION));
+  unsigned long number_of_points = ceil(length / (2 * RESOLUTION));
```
Comment (ooeygui) [inline nav2_dwb_controller/nav_2d_utils/test/path_ops_test.cpp]: uint64_t

File: nav2_map_server/test/component/CMakeLists.txt [hunk 4]
```diff
@@ -30,7 +29,6 @@ ament_add_gtest_executable(test_map_saver_node
 ament_target_dependencies(test_map_saver_node rclcpp nav_msgs)
 target_link_libraries(test_map_saver_node
   ${library_name}
-  stdc++fs
 )
```
Comment (ooeygui) [inline nav2_map_server/test/component/CMakeLists.txt:37]: if (NOT WIN32)
stdc++fs
endif()

File: nav2_waypoint_follower/include/nav2_waypoint_follower/plugins/photo_at_waypoint.hpp [hunk 5]
```diff
@@ -32,8 +32,8 @@
 
 #include "sensor_msgs/msg/image.hpp"
 #include "nav2_core/waypoint_task_executor.hpp"
-#include "opencv4/opencv2/core.hpp"
-#include "opencv4/opencv2/opencv.hpp"
+#include "opencv2/core.hpp"
```
Comment (ooeygui) [inline nav2_waypoint_follower/include/nav2_waypoint_follower/plugins/photo_at_waypoint.hpp:40]: #ifdef WIN32
#else
opencv4/opencv
#endif

**Review by ooeygui — CHANGES_REQUESTED** [review]

File: nav2_map_server/test/component/CMakeLists.txt (hunk 4 above)
Comment (PranavDhulipala) [reply nav2_map_server/test/component/CMakeLists.txt:37]: @ooeygui this is required for experimental/filesystem, wouldn't this be redundant when using  c++ 17?

File: nav2_smac_planner/CMakeLists.txt [hunk 6]
```diff
@@ -92,6 +94,10 @@ add_library(${library_name} SHARED
 
 target_link_libraries(${library_name} ${OMPL_LIBRARIES} ${OpenMP_LIBRARIES}  OpenMP::OpenMP_CXX)
 target_include_directories(${library_name} PUBLIC ${Eigen3_INCLUDE_DIRS})
+if(WIN32)
```
Comment (ooeygui) [inline nav2_smac_planner/CMakeLists.txt]: This should be set on both linux and windows, no need to wrap in WIN32

File: nav2_smac_planner/test/CMakeLists.txt [hunk 7]
```diff
@@ -115,4 +115,10 @@ ament_add_gtest(test_lattice_node test_nodelattice.cpp)
 
 ament_target_dependencies(test_lattice_node ${dependencies})
 
-target_link_libraries(test_lattice_node ${library_name})
+target_link_libraries(test_lattice_node ${library_name} 
+  ${library_name}_lattice
+)
+# if(WIN32)
```
Comment (ooeygui) [inline nav2_smac_planner/test/CMakeLists.txt]: Commented out code? Should it be here and enabled or removed?

**Review by ooeygui — CHANGES_REQUESTED** [review]

File: nav2_smac_planner/test/CMakeLists.txt (hunk 7 above)
Comment (PranavDhulipala) [reply nav2_smac_planner/test/CMakeLists.txt]: this needs to be removed, made the change

File: nav2_smac_planner/CMakeLists.txt (hunk 6 above)
Comment (PranavDhulipala) [reply nav2_smac_planner/CMakeLists.txt]: fixed this

**Pull request merged**